vhost-device-vsock/src/main.rs

```diff
@@ -173,11 +177,11 @@ fn parse_vm_params(s: &str) -> Result<VsockConfig, VmArgsParseError> {
 impl VsockArgs {
     pub fn parse_config(&self) -> Option<Result<Vec<VsockConfig>, CliError>> {
         if let Some(c) = &self.config {
-            let b = config::Config::builder()
-                .add_source(config::File::new(c.as_str(), config::FileFormat::Yaml))
-                .build();
-            if let Ok(s) = b {
-                let mut v = s.get::<Vec<ConfigFileVsockParam>>("vms").unwrap();
+            let figment = Figment::new().merge(Yaml::file(c.as_str()));
+
+            let config_map = figment.extract::<HashMap<String, Vec<ConfigFileVsockParam>>>();
+            if let Ok(mut s) = config_map {
+                let v = s.get_mut("vms").unwrap();
```

Comment:
It's pre-existing, but since we are here, can we clean it a bit and remove `s` (which I don't remember what meant). Something like this:

```diff
diff --git a/vhost-device-vsock/src/main.rs b/vhost-device-vsock/src/main.rs
index 20a1ebb..ec2d77e 100644
--- a/vhost-device-vsock/src/main.rs
+++ b/vhost-device-vsock/src/main.rs
@@ -179,9 +179,10 @@ impl VsockArgs {
         if let Some(c) = &self.config {
             let figment = Figment::new().merge(Yaml::file(c.as_str()));
 
-            let config_map = figment.extract::<HashMap<String, Vec<ConfigFileVsockParam>>>();
-            if let Ok(mut s) = config_map {
-                let v = s.get_mut("vms").unwrap();
+            if let Ok(mut config_map) =
+                figment.extract::<HashMap<String, Vec<ConfigFileVsockParam>>>()
+            {
+                let v = config_map.get_mut("vms").unwrap();
                 if !v.is_empty() {
                     let parsed: Vec<VsockConfig> = v
                         .drain(..)
```

Reply:
ok makesense, I think we can also do thesame for `v` maybe replace it `vms_param` WDYT?

Reply:
> ok makesense, I think we can also do thesame for `v` maybe replace it `vms_param` WDYT?

for `v` we need to touch more code, so I'd do in a different patch, but you can include it in this PR if you want.
Maybe we can do a first patch that just fixes the variable names and second patch that replace the crate.

Reply:
Thank you! I split it.